usr/bs_hyc.c

```diff
@@ -136,16 +152,19 @@ static int bs_hyc_cmd_submit(struct scsi_cmd *cmdp)
 		reqid = HycScheduleWrite(infop->rpc_con, cmdp, bufp, length, offset);
 		break;
 	case WRITE_SAME_OP:
-		/** TODO */
 	case UNKNOWN:
 	default:
 		assert(0);
 	}
 
 	/* If we got reqid, set it in hyc_cmd */
 	if (hyc_unlikely(reqid == kInvalidRequestID)) {
-		eprintf("request submission got err invalid request\n");
-		target_cmd_io_done(cmdp, SAM_STAT_CHECK_CONDITION);
+		eprintf("request submission got error invalid request" 
+			" size: %lu offset : %"PRIu64" opcode :%u\n", 
+			length, offset, (unsigned int) cmdp->scb[0]);
```

Comment:
We need to test this scenario.. Set the Reqid invalid for write IOs (don't do it for read IOs because that will fail discovery) and see wherther application (may be "dd" command ) is getting completion(error) or not.